### PR TITLE
#8581 mod listing page object model and organizing page editor POMs

### PR DESCRIPTION
**Pull request by fungairino**
## What does this PR do?

It introduces a new file `modListingPanel.ts` which contains classes for handling the mod listing panel in the page editor. It also moves the main page editor POM to its own directory. The changes are part of the ongoing effort to improve the maintainability and readability of the end-to-end testing codebase.

Part of #8581

### CODE REVIEW

File: knip.mjs [hunk 1]
```diff
@@ -45,6 +45,8 @@ const knipConfig = {
     "src/development/hooks/**",
     // Type-only strictNullChecks helper
     "src/types/typeOnlyMessengerRegistration.ts",
+    // End-to-end tests
```
Comment (mnholtz) [inline knip.mjs:48]: nit: unnecessary comment imo 😅

File: end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { BasePageObject } from "../basePageObject";
+import { uuidv4 } from "@/types/helpers";
+import { type Locator } from "@playwright/test";
+
+// Starter brick names as shown in the Page Editor UI
+export type StarterBrickName =
```
Comment (mnholtz) [inline end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts:23]: To get rid of comment?
```suggestion
export type StarterBrickUIName =
```

File: end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  | "Trigger"
+  | "Button"
+  | "Quick Bar Action"
+  | "Dynamic Quick Bar"
+  | "Sidebar Panel";
+
+export class ModListItem extends BasePageObject {
+  saveButton = this.locator("[data-icon=save]");
+  get menuButton() {
+    return this.getByLabel(`${this.modComponentName} - Ellipsis`);
```
Comment (mnholtz) [inline end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts:34]: nice, I didn't realize we had that aria label on those buttons

**Review by mnholtz — COMMENTED** [review]
looks good so far; lmk when you resolve failing tests

File: end-to-end-tests/pageObjects/extensionConsole/modsPage.ts [hunk 4]
```diff
@@ -38,9 +38,10 @@ export class ModsPage extends BasePageObject {
     // TODO: remove once fixed: https://github.com/pixiebrix/pixiebrix-extension/issues/8458
     const registryPromise = this.page
       .context()
-      .waitForEvent("requestfinished", (request) =>
-        request.url().includes("/api/registry/bricks/"),
-      );
```
Comment (fungairino) [inline end-to-end-tests/pageObjects/extensionConsole/modsPage.ts:43]: not directly related to this PR, but I had to add this timeout after defining a default timeout for actions

File: playwright.config.ts [hunk 5]
```diff
@@ -37,6 +37,12 @@ export default defineConfig<{ chromiumChannel: string }>({
 
     /* Collect trace when retrying the failed test in CI, and always on failure when running locally. See https://playwright.dev/docs/trace-viewer */
     trace: CI ? "on-first-retry" : "retain-on-failure",
+
+    /* Set the default timeout for actions such as `click` */
+    actionTimeout: 5_000,
+
+    /* Set the default timeout for page navigations */
+    navigationTimeout: 10_000,
```
Comment (fungairino) [inline playwright.config.ts:45]: waiting the whole test timeout for a failed click action is annoying so I defined a default action and nav timeouts.

**Review by mnholtz — APPROVED** [review]

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**